src/kernl/model_optimization.py

```diff
@@ -12,46 +12,50 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 #
-from typing import Callable, List
+from typing import List
 
 import torch
-import torchdynamo
+import torch._dynamo as torchdynamo
 from transformers import PreTrainedModel
 
 from kernl.implementations.cuda_graph import cuda_graphs_wrapper
 from kernl.optimizer.dynamo_backend import dynamo_backend_ofi
 
 
-def optimize_model(original_model: PreTrainedModel, pool: (int, int) = torch.cuda.graph_pool_handle()) -> Callable:
+# single shared pool by default
+_pool: (int, int) = torch.cuda.graph_pool_handle()
+
+
+# needs to be generated once to be reused several times, like encoder/decoder models
+# https://github.com/pytorch/torchdynamo/issues/1816
+def _compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
+    dynamo_backend_ofi(gm)
+    return cuda_graphs_wrapper(gm, example_inputs, pool=_pool)
+
+
+def optimize_model(original_model: PreTrainedModel) -> None:
     """
     Optimizes a given model. Optimization is done in two steps:
     *  first step is to convert the given model to fx graph.
     *  second step is to replace patterns found in the graph in order to optimize the model.
+    Example:
+    model = AutoModel.from_pretrained(...).eval().cuda()
 
-    @return: returns the optimized model (and the original model is modified, so it can not be used after optimization).
-
-        Example:
-            from kernl.model_optimization import optimize_model
+    optimize_model(model)
+    inputs = ...
+    model(**inputs)
 
-            model_name = "BaptisteDoyen/camembert-base-xnli"
-            model = AutoModelForSequenceClassification.from_pretrained(model_name)
-            model = model.eval().cuda()
-
-            optimized_model = optimize_model(model)
+    @param original_model: model to optimize
```

Comment:
return missing, and description must indicate that is does a mutation of the model

Reply:
there is no return so it makes it obvious there is a mutation. I will add the point to the doc.

Reply:
with this new call, we break the api, we'll have to change the major version after that. I think our main goal in the future may be to have a method that does not mutate the original model (so a new breaking change in the api). Is it OK or we may work more on the API in order not to break easily next time ?